src/components/ShowWeapons.jsx

```diff
@@ -26,7 +26,11 @@ class ShowWeapons extends React.Component {
             onClick={(event) => {
               const newSelected = !this.state.selectGists;
               this.setState({ selectGists: newSelected });
-            }}
+              if ({ selectGists: true }) {
```

Comment:
an non empty object is always true. I guess what you want to write is:

    if (this.state.selectGists === true) {